source/core/DualTypeVisitor.cpp

```diff
@@ -29,97 +29,99 @@ namespace Vireo
     bool DualTypeVisitor::TypesAreCompatible(TypeRef typeRefX, void* pDataX, TypeRef typeRefY, void* pDataY, const DualTypeOperation &operation)
     {
         bool success = false;
-        if (typeRefX->IsVariant() && typeRefY->IsVariant()) {
-            success = VariantCompatible(typeRefX, pDataX, typeRefY, pDataY, operation);
-        } else {
-            EncodingEnum encodingX = typeRefX->BitEncoding();
-            switch (encodingX) {
-            case kEncoding_Boolean:
-                success = operation.AreBooleanCompatible(typeRefX, typeRefY);
-                break;
-            case kEncoding_UInt:
-                success = operation.AreUIntCompatible(typeRefX, typeRefY);
-                break;
-            case kEncoding_S2CInt:
-                success = operation.AreS2CIntCompatible(typeRefX, typeRefY);
-                break;
-            case kEncoding_IEEE754Binary:
-                success = operation.AreIEEE754BinaryCompatible(typeRefX, typeRefY);
-                break;
-            case kEncoding_Cluster:
-                success = typeRefY->BitEncoding() == kEncoding_Cluster && ClusterCompatible(typeRefX, pDataX, typeRefY, pDataY, operation);
-                break;
-            case kEncoding_Enum:
-                success = EnumCompatible(typeRefX, typeRefY, operation);
-                break;
-            case kEncoding_Array: {
-                if (typeRefX->Rank() == 1 && typeRefX->GetSubElement(0)->BitEncoding() == kEncoding_Unicode) {
-                    success = StringCompatible(typeRefX, typeRefY);
-                } else {
-                    success = typeRefY->BitEncoding() == kEncoding_Array && ArrayCompatible(typeRefX, pDataX, typeRefY, pDataY, operation);
-                }
-                break;
-            }
-            case kEncoding_None:
```

Comment:
Adding a new encoding type per @spathiwa 's advice cleaned this up a lot in using the encoding type for variants. Thanks a lot Craig.